packages/compiler/src/core/checker.ts

```diff
@@ -2500,6 +2515,38 @@ export function createChecker(program: Program): Checker {
     }
   }
 
+  function checkModelValidates(
+    node: ModelStatementNode | ScalarStatementNode,
+    validates: Map<string, ModelValidate>,
+    parentModel: Model | Scalar,
+    mapper: TypeMapper | undefined
+  ) {
+    for (const vv of node.validates) {
+      const newV = checkModelValidate(vv, mapper);
+      newV.model = parentModel;
+      defineValidator(validates, newV);
+    }
+  }
+
+  function defineValidator(
+    validates: Map<string, ModelValidate>,
+    newV: ModelValidate,
+    diagnosticTarget?: DiagnosticTarget
+  ) {
+    if (validates.has(newV.name)) {
+      reportCheckerDiagnostic(
+        createDiagnostic({
+          code: "duplicate-validate",
+          format: { propName: newV.name },
+          target: diagnosticTarget ?? newV,
+        })
+      );
+      return;
+    }
+
+    validates.set(newV.name, newV);
```

Comment:
Does this work when there are multiple anonymous validates expressions? For unions we use a fresh symbol for anonymous properties, which might work here too.

Reply:
Good catch, fixed this!